hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -272,11 +278,12 @@ private StreamingRecordIterator(Schema readerSchema, Schema writerSchema, Seekab
         this.totalRecords = this.inputStream.readInt();
       }
 
-      if (recordNeedsRewriteForExtendedAvroTypePromotion(writerSchema, readerSchema)) {
-        this.reader = new GenericDatumReader<>(writerSchema, writerSchema);
+      Schema repairedWriterSchema = AvroSchemaRepair.repairLogicalTypes(writerSchema, readerSchema);
```

Comment:
should we also check the flag `enableLogicalTimestampFieldRepair` ?

Reply:
We're prioritizing correctness in this PR.  The schema repair overhead is minimal.  We'll take the minor perf optimization separately.